pinecone/config/config.py

```diff
@@ -41,6 +42,7 @@ def build(
         api_key = api_key or kwargs.pop("api_key", None) or os.getenv("PINECONE_API_KEY")
         host = host or kwargs.pop("host", None)
         host = normalize_host(host)
+        source_partner = kwargs.pop("source_partner", None)
```

Comment:
Currently I'm thinking this would go in `kwargs` vs a named parameter as this would be more of an internal field we'd just ask partners to include in their integrations.

Reply:
I agree `kwargs` makes sense here because we don't want to invite randos to set strange values. Perhaps partner/integration name and version should be passed separately so that we can have greater control over formatting.  For your example, perhaps I would prefer `pc = Pinecone(api_key='foo', source_id='bar', source_version='1.2')`. That way you don't end up with one person passing `foo/1.2` and another passing `bar 3.5` and causing headaches in parsing when it's time to try to turn this all into some kind of dashboard.

Reply:
Agree on separating the name from version.

Reply:
This is done in the latest version